m_monit_adapter.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'status/server'
+require 'mmonit'
+
+class MMonitAdapter
+
+  attr_accessor :mmonit
+
+  def initialize(username, password, address, port)
+    @username = username
+    @password = password
+    @address = address
+    @port = port
+  end
+
+  def connect
+    @mmonit ||= MMonit::Connection.new(
+        ssl: false,
+        username: @username,
+        password: @password,
+        address: @address,
+        port: @port
+    )
+    @mmonit.connect
+  end
+
+  def disconnect
+    if mmonit
+      mmonit.disconnect
+    end
+  end
```

Comment:
Deze roep je nu zo aan:
```ruby
adapter.connect
@servers = adapter.statuses
adapter.disconnect
```
Maar als er iets misgaat in `adapter.statuses`, dan wordt `#disconnect` nooit aangeroepen.

Je kunt in de aanroepende code een `ensure` toevoegen:
```ruby
begin
  adapter.connect
  @servers = adapter.statuses
ensure
  adapter.disconnect
end
```
maar je kunt hem ook hier aanpakken door zoiets te doen:
```ruby
class MMonitAdapter
  def connect
    @mmonit ||= MMonit::Connection.new(
      ssl: false,
      username: @username,
      password: @password,
      address: @address,
      port: @port
    )
  
    @mmonit.connect
  
    yield self
  ensure
    if @mmonit
      @mmonit.disconnect
      @mmonit = nil
    end
  end
end
```